x/gamm/pool-models/stableswap/pool.go

```diff
@@ -69,6 +69,31 @@ func (pa Pool) getPoolAmts(denoms ...string) ([]sdk.Int, error) {
 	return result, nil
 }
 
+func (pa Pool) getScaledPoolAmts(denoms ...string) ([]sdk.Int, error) {
+	result := make([]sdk.Int, len(denoms))
+	poolLiquidity := pa.PoolLiquidity
+	for i, d := range denoms {
+		amt := poolLiquidity.AmountOf(d)
+		if amt.IsZero() {
+			return []sdk.Int{}, fmt.Errorf("denom %s does not exist in pool", d)
+		}
+
+		scalingFactor := pa.getPoolAssetScalingFactor(d)
+		result[i] = amt.QuoRaw(int64(scalingFactor))
+	}
+	return result, nil
+}
+
+func (pa Pool) getPoolAssetScalingFactor(denom string) uint64 {
+	var result uint64
+	for _, scalingFactor := range pa.PoolScalingFactor {
```

Comment:
Good catch!

Reply:
We should require them to be in the same order by construction, then we can just use the same index over both, right?

Reply:
Both options look like a valid option! 

I'd prefer @ValarDragon's option to simply use the same index rather than changing poolLiquidity(which is a simply sdk.Coin field) to a separate struct of it's own!

Reply:
Switching to @p0mvn's proposal. Even though we try to use the same index for both, we would still end up with O(n^2), since we would have to iterate over given denoms, then do another iteration to find the poolLiquidity index.

Reply:
This can be changed to be O(N) in software, no serialization changes, with mild complexity using a map. `O(n^2)` is fine for an initial version